### PR TITLE
Adding gtest and gmock third party libraries. Modified premake config file. Added new test project for Graphics module.

### PR DESCRIPTION
**Pull request by travisSVK**

### CODE REVIEW

File: Source/Graphics/UnitTest/GraphicsTest.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    {
+
+        class GraphicsTestBase : public ::testing::Test {
+
+        protected:
+
+            /*
+            * Define ctor for to initialize for whole duration of test.
+            */
+            // ~GraphicsTestBase(){}
```
Comment (ludvigarlebrink) [inline Source/Graphics/UnitTest/GraphicsTest.cpp]: Why commented code?

File: Source/Graphics/UnitTest/GraphicsTest.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#include <gtest/gtest.h>
+#include <memory>
+#include "../RenderWindow.hpp"
+
+namespace gdt
+{
+    namespace test
+    {
+
+        class GraphicsTestBase : public ::testing::Test {
```
Comment (ludvigarlebrink) [inline Source/Graphics/UnitTest/GraphicsTest.cpp]: Document the class.

File: Source/Graphics/UnitTest/GraphicsTest.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+* @date     9/16/2018
+*
+* @brief Graphics module tests
+*/
+
+#include <gtest/gtest.h>
+#include <memory>
+#include "../RenderWindow.hpp"
+
+namespace gdt
```
[the remark this answers is not in the document]
Comment (travisSVK) [reply Source/Graphics/UnitTest/GraphicsTest.cpp]: Ok we can discuss the namespace structure.

File: Source/Graphics/UnitTest/GraphicsTest.cpp (hunk 1 above)
Comment (travisSVK) [reply Source/Graphics/UnitTest/GraphicsTest.cpp]: Not needed now, using it just for example. Default ctor and dtor are enough in this class as is right now.

**Comment by travisSVK** [comment]
Changes commited.

**Pull request merged**